Update build info

sithlord48 opened the pull request:
- Add a page about the build options [preview](https://github.com/sithlord48/wiki-prs/blob/update_BuildInfo/wiki/Build-Options.md)
 - Update Mac os build guide [preview](https://github.com/sithlord48/wiki-prs/blob/master/wiki/Building-on-macOS.md)
 - Update Linux Build Guide [preview](https://github.com/sithlord48/wiki-prs/blob/update_BuildInfo/wiki/Building-on-Linux.md)
 -  Update Windows Build Guide [preview](https://github.com/sithlord48/wiki-prs/blob/update_BuildInfo/wiki/Building-on-Windows.md)

Matching code changes https://github.com/input-leap/input-leap/pull/1981

sithlord48 committed:
docs: Update Mac Os Build Documentation

sithlord48 committed:
Add Page Describing Build Options

sithlord48 committed:
docs: Update Linux build documentation

AlanMuha commented on wiki/Building-on-Linux.md:
```diff
@@ -7,27 +7,32 @@
 sudo apt install git cmake make xorg-dev g++ libcurl4-openssl-dev \
                  libavahi-compat-libdnssd-dev libssl-dev libx11-dev \
-                 qttools5-dev qtbase5-dev libice-dev libsm-dev
+                 qttools6-dev qtbase6-dev libice-dev libsm-dev
```
Regarding the Ubuntu/Debian/Raspbian equivalent packages for `qttools5-dev` and `qtbase5-dev`, seem as if the actual equivalent (at least when I was trying to install from source in my Raspbian system) is `qt6-tools-dev` and `qt6-base-dev`

sithlord48 replied:
Thank you for the input , updated.

KoljaFrahm commented on wiki/Building-on-Linux.md:
```diff
@@ -7,27 +7,32 @@
                  libcurl-devel avahi-compat-libdns_sd-devel \
-                 libXtst-devel qt5-qtbase qt5-qtbase-devel  \
-                 qt5-qttools-devel libICE-devel libSM-devel \
+                 libXtst-devel qt6-qtbase qt6-qtbase-devel  \
```
Doesn't need qt6-qtbase
libei-devel and libportal-devel should be added

KoljaFrahm commented on wiki/Building-on-macOS.md:
````diff
@@ -4,76 +4,59 @@
+  brew install qt cmake openssl git pkg-config
   ```
 
-### Clone the repository
````
Why delete this part?

KoljaFrahm commented on wiki/Building-on-Linux.md:
```diff
@@ -40,26 +45,38 @@ sudo dnf groupinstall "Development Tools"
 sudo zypper install libdrm-devel libglvnd-devel libICE-devel \
-        libQt5Core-devel libQt5Gui-devel libQt5Network-devel \
-        libqt5-qtbase-common-devel libQt5Widgets-devel libSM-devel \
+        libQt6Core-devel libQt6Gui-devel libQt6Network-devel \
```
Are your sure these are the right packages? I don't think they exist

KoljaFrahm replied:
pkgconfig(Qt6Core)
pkgconfig(Qt6Gui)
pkgconfig(Qt6Network)
pkgconfig(Qt6Platform)
pkgconfig(Qt6Widgets)

I think it also needs pkgconfig(Qt6Linguist)

KoljaFrahm replied:
How about replacing most by pkgconfig?

KoljaFrahm commented on wiki/Build-Options.md:
```diff
@@ -0,0 +1,13 @@
+QT_DEFAULT_MAJOR_VERSION       | Version of Qt to use for gui  (5 or 6)         | 6                  | Must Build Gui For any Effect |
+INPUTLEAP_BUILD_TESTS          | Builds inputleap tests                         | ON                 | |
+INPUTLEAP_BUILD_EXTERNAL_GTEST | Use System Google Test for test building       | OFF                | Google Tests on your system |
+INPUTLEAP_BUILD_X11            | XWindows support (required) (linux only        | ON                 | xorg-devel |
```
) missing

KoljaFrahm commented on wiki/Building-on-Linux.md:
````diff
@@ -40,26 +45,38 @@ sudo dnf groupinstall "Development Tools"
 ```
 
-### openSUSE
+### openSUSE## Building the Project
````
why does it have building the project here?

KoljaFrahm commented on wiki/Building-on-Linux.md:
````diff
@@ -40,26 +45,38 @@ sudo dnf groupinstall "Development Tools"
 ## Building from source
 
-```shell
-git clone https://github.com/input-leap/input-leap.git
````
why remove this?

AlanMuha reviewed (commented):
Maybe specify on the Linux Build that Qt6 is the default option, in the dependency list. Also in Ubuntu section, it still has qt5 dependencies, not qt6s (`qt6-tools-dev`, `qt6-base-dev`)

sithlord48 committed:
docs: Update Windows build documentation

sithlord48 commented:
> Maybe specify on the Linux Build that Qt6 is the default option, in the dependency list. Also in Ubuntu section, it still has qt5 dependencies, not qt6s (`qt6-tools-dev`, `qt6-base-dev`)

Added the note that Qt6 is default and  removed the ubuntu package install line for now as no release has the libei or libportal without building them.

KoljaFrahm commented on wiki/Building-on-Linux.md:
```diff
@@ -7,59 +15,65 @@
 **Note**: these aren't the only things built at compile time, but merely what the end-user is expected to use.
 
 ## Preparing your system
+This project depends upon the following Make sure you install these before you build. If you get lost you can always check the [github workflow](https://github.com/input-leap/input-leap/blob/master/.github/workflows/builds.yml) to see how we build for your distro. Depending on your distrobution you may need to install `dev` or `devel` pacakges to get the files need to build aginst the dependencie ...
```
remove "This project depends upon the following"
pacakges
aginst

KoljaFrahm replied:
Otherwise Linux section lgtm

sithlord48 replied:
Should be good now thanks
